wallet-new/src/Cardano/Wallet/Kernel/Accounts.hs

```diff
@@ -0,0 +1,208 @@
+      -- ^ When trying to create the 'Account', the parent 'HdRoot' was not
+      -- there.
+    | CreateAccountKeystoreNotFound WalletId
+      -- ^ When trying to create the 'Account', the 'Keystore' didn't have
+      -- any secret associated with the input 'WalletId'.
+    | CreateAccountHdRndAccountSpaceSaturated HdRootId
+      -- ^ The available number of HD accounts in use is such that trying
+      -- to find another random index would be too expensive.
+    deriving Eq
+
+instance Buildable CreateAccountError where
+    build (CreateAccountUnknownHdRoot uRoot) =
+        bprint ("CreateAccountUnknownHdRoot " % F.build) uRoot
+    build (CreateAccountKeystoreNotFound accId) =
+        bprint ("CreateAccountKeystoreNotFound " % F.build) accId
+    build (CreateAccountHdRndAccountSpaceSaturated hdAcc) =
+        bprint ("CreateAccountHdRndAccountSpaceSaturated " % F.build) hdAcc
+
+instance Show CreateAccountError where
+    show = formatToString build
+
+instance Exception CreateAccountError
+
+--
+--  Delete-related errors
+--
+
+data DeleteAccountError =
+      DeleteAccountUnknownHdRoot HdRootId
+      -- ^ When trying to delete the 'Account', the parent 'HdRoot' was not
+      -- there.
+    deriving Eq
+
+instance Buildable DeleteAccountError where
+    build (DeleteAccountUnknownHdRoot uRoot) =
+        bprint ("DeleteAccountUnknownHdRoot " % F.build) uRoot
+
+instance Show DeleteAccountError where
+    show = formatToString build
+
+instance Exception DeleteAccountError
+
+--
+-- Update-related errors
+
+data UpdateAccountError =
+      UpdateAccountUnknownHdRoot HdRootId
+      -- ^ When trying to update the 'Account', the parent 'HdRoot' was not
+      -- there.
+    | UpdateAccountUnknownHdAccount HdAccountId
+      -- ^ When trying to update the 'Account', the account was not there.
+    deriving Eq
+
+instance Buildable UpdateAccountError where
+    build (UpdateAccountUnknownHdRoot uRoot) =
+        bprint ("UpdateAccountUnknownHdRoot " % F.build) uRoot
+    build (UpdateAccountUnknownHdAccount uAccount) =
+        bprint ("UpdateAccountUnknownHdAccount " % F.build) uAccount
+
+instance Show UpdateAccountError where
+    show = formatToString build
+
+instance Exception UpdateAccountError
+
+-- | Creates a new 'Account' for the input wallet.
+-- Note: @it does not@ generate a new 'Address' to go in tandem with this
+-- 'Account'. This will be responsibility of the wallet layer.
+createAccount :: PassPhrase
+              -- ^ The 'Passphrase' (a.k.a the \"Spending Password\").
+              -> AccountName
+              -- ^ The name for this account.
+              -> WalletId
+              -- ^ An abstract notion of a 'Wallet identifier
+              -> PassiveWallet
+              -> IO (Either CreateAccountError HdAccount)
+createAccount spendingPassword accountName walletId pw = do
+    let keystore = pw ^. walletKeystore
+    case walletId of
+         (WalletIdHdRnd hdRootId) -> do
```

Comment:
You loooove brackets :D

Reply:
What can I say, I used to program in Common Lisp :trollface: . Will fix.